src/jit/lowerxarch.cpp

```diff
@@ -2299,6 +2299,18 @@ void Lowering::TreeNodeInfoInitLogicalOp(GenTree* tree)
         // as reg optional.
         SetRegOptionalForBinOp(tree);
     }
+
+    if (varTypeIsByte(op1))
+    {
+        regMaskTP regMask;
+        bool      containedNode = (op2->gtLsraInfo.srcCount == 0) && (op2->gtLsraInfo.dstCount == 0);
+        if (!containedNode)
+        {
+            regMask = op2->gtLsraInfo.getSrcCandidates(l);
+            assert(regMask != RBM_NONE);
+            op2->gtLsraInfo.setSrcCandidates(l, regMask & ~RBM_NON_BYTE_REGS);
+        }
```

Comment:
I think the right condition to check for logical operators (add,sub, and, or, xor) is

if (varTypeIsByte(op1) || varTypeIsByte(op2))
{
     // Set source candidates of op1 and op2 to be byteable registers
     // Set destination candidates of tree to be byteable registers
}

Note that the above logic would be more restrictive than required in the following case
GT_ADD(op1= byte ptr mem-op, op2=TYP_INT or TYP_BYTE in a reg)

We just need to constrain dst candiates of GT_ADD to byteable regs and not op2.
During codegen, if op2's Reg == targetReg, then op2 is already in a byteable reg.
If op2's reg != targetReg,  codegen will do a reg-to-reg move using genActualType(targetType).